extensions/ql-vscode/src/run-queries.ts

```diff
@@ -612,6 +617,24 @@ export async function compileAndRunQueryAgainstDatabase(
     void logger.log(`Couldn't resolve metadata for ${qlProgram.queryPath}: ${e}`);
   }
 
+  let availableMlModels: cli.MlModelInfo[] = [];
+  // The `capabilities.untrustedWorkspaces.restrictedConfigurations` entry in package.json doesn't
+  // work with hidden settings, so we manually check that the workspace is trusted before looking at
+  // whether the `shouldInsecurelyLoadMlModelsFromPacks` setting is enabled.
```

Comment:
Arguably, we should _never_ be insecurely loading anything from an untrusted workspace. So, I think this makes sense regardless.

Reply:
Though, I am a little confused. It looks like unless `shouldInsecurelyLoadMlModelsFromPacks` is true, there will be _no_ ml models available. 

Although, perhaps what's going on is that this command only loads extra ml models not available in the default distro. And default ml models will always be available.

Reply:
The `shouldInsecurelyLoadMlModelsFromPacks` setting allows internal users to turn on loading ML models from CodeQL packs.  While it's in beta, we assume that this is an insecure operation, hence the `insecurely`.  However once out of beta we plan for this operation to be a secure one, allowing us to load a limited set of GitHub-authored ML models by default and removing the need for the setting.

Reply:
If you want to be extra defensive, you could also insist that the canary flag is enabled.